contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -138,29 +153,30 @@ contract ItemsRolesRegistryFacet is Modifiers, ISftRolesRegistry, ERC1155Holder
         );
     }
 
-    function withdraw(
-        uint256 _nonce
-    )
-        public
-        onlyOwnerOrApprovedWithBalance(
-            s.itemsDeposits[_nonce].grantor,
-            s.itemsDeposits[_nonce].tokenAddress,
-            s.itemsDeposits[_nonce].tokenId,
-            s.itemsDeposits[_nonce].tokenAmount
-        )
-    {
-        DepositInfo memory _depositInfo = s.itemsDeposits[_nonce];
-        require(_depositInfo.tokenAmount > 0, "ItemsRolesRegistryFacet: deposit does not exist");
+    function _unequipWearable(uint256 _gotchiId, uint256 _wearableToUnequip) internal {
+        for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            if (s.aavegotchis[_gotchiId].equippedWearables[slot] != _wearableToUnequip) continue;
+            s.aavegotchis[_gotchiId].equippedWearables[slot] = 0;
+        }
```

Comment:
Is the loop really necessary?

Reply:
yes, we don't know in which slot the delegated wearable was equipped.